### PR TITLE
2.1 capability

### PR DESCRIPTION
**Pull request by rpiazza**
changes before testing with validator

### CODE REVIEW

File: stix2elevator/confidence.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+def convert_numeric_string(value):
+    if value.find(".") == -1:
+        return int(value)
+    else:
+        return float(value)
+
+
+def convert_confidence_value(value, id):
+    if isinstance(value, int) or isinstance(value, long):
```
[the remark this answers is not in the document]
Comment (rpiazza) [reply stix2elevator/confidence.py]: Done

File: stix2elevator/confidence.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            warn("The confidence value %s in %s has been converted to an integer so it is valid in STIX 2.0", 724,
+                 value, id)
+            confidentiality2_1_value = ceil(value)
+    elif isinstance(value, str):
+        value = text_type(value)
+        if value.isnumeric():
+            confidentiality2_1_value = convert_confidence_value(convert_numeric_string(value), id)
+        else:
+            confidentiality2_1_value = convert_confidence_string(value)
+    elif isinstance(value, object):
```
[the remark this answers is not in the document]
Comment (rpiazza) [reply stix2elevator/confidence.py]: Not sure what you are suggesting here....

File: stix2elevator/convert_cybox.py [hunk 3]
```diff
@@ -234,7 +237,7 @@ def convert_process(process):
     process_dict = {"type": "process"}
     cybox_dict[text_type(index)] = process_dict
     index += 1
-    if process.name:
+    if process.name and get_option_value("spec_version") == "2.0":
```
[the remark this answers is not in the document]
Comment (rpiazza) [reply stix2elevator/convert_cybox.py]: I'm not sure these are the same.  The image_ref.name is the name of the executable file, which may not be the name of the process

File: stix2elevator/convert_cybox.py [hunk 4]
```diff
@@ -247,7 +250,7 @@ def convert_process(process):
     if process.parent_pid:
         create_process_ref(process.parent_pid, process_dict, cybox_dict, index, "parent_ref")
         index += 1
-    if process.argument_list:
+    if process.argument_list and get_option_value("spec_version") == "2.0":
```
[the remark this answers is not in the document]
Comment (rpiazza) [reply stix2elevator/convert_cybox.py]: Let me look into this one.

File: stix2elevator/convert_cybox.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (clenk) [reply stix2elevator/convert_cybox.py]: Ok, that's valid.

File: stix2elevator/confidence.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (clenk) [reply stix2elevator/confidence.py]: Ignore that comment - misread some indentation. :P

File: stix2elevator/convert_pattern.py [hunk 5]
```diff
@@ -43,6 +47,101 @@
...
+        current_cyber_observable_type = self.object_type_name
+        for x in self.property_path:
+            if x.property_name == "extensions":
+                continue
+            if current_cyber_observable_type == "file":
+                if (x.property_name == "is_encrypted" or
+                        x.property_name == "encryption_algorithm" or
+                        x.property_name == "decryption_key"):
+                    print(
+                        "Expression contains the property " + x.property_name + ", for a file, which is not in STIX 2.1")
```
[the remark this answers is not in the document]
Comment (rpiazza) [reply stix2elevator/convert_pattern.py]: Sounds reasonable - I'll look into it

**Comment by clenk** [comment]
Thank you @rpiazza!

**Pull request merged**